docs/howto/cli-completion.rst

```diff
@@ -0,0 +1,42 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Set Up Bash/Zsh Completion
+==========================
+
+When using bash (or ``zsh``) as your shell, ``airflow`` can use
+`argcomplete <https://argcomplete.readthedocs.io/>`_ for auto-completion.
+
+For global activation of all argcomplete enabled python applications run:
+
+.. code-block:: bash
+
+  sudo activate-global-python-argcomplete
+
+For permanent (but not global) airflow activation, use:
+
+.. code-block:: bash
+
+  register-python-argcomplete airflow >> ~/.bashrc
+
+For one-time activation of argcomplete for airflow only, use:
+
+.. code-block:: bash
+
+  eval "$(register-python-argcomplete airflow)"
```

Comment:
Is it possible to have this as `airflow autocompletions` or something (which just tries to import that module etc.)

Reply:
Not sure as it would have to find out current Shell (bash/zsh - which might be easy) and then export that command to `.bash_rc` or `.bash_profile` or `.zshrc` etc by running some bash commands?

Reply:
Just the `register-python-argcomplete airflow` bit - not the eval etc.

So this line would be

```suggestion
  eval "$(airflow autocompletions)"
```

Reply:
Aah ok. That should be plausible. Let me give it a go

Reply:
Well this isn't as trivial as I thought it would because just running `airflow` prints out the following:

```
/Users/kaxilnaik/Documents/GitHub/incubator-airflow/airflow/models/dagbag.py:21: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
[2019-08-13 22:30:04,803] {settings.py:173} INFO - settings.configure_orm(): Using pool settings. pool_size=5, max_overflow=10, pool_recycle=1800, pid=86597
/Users/kaxilnaik/.virtualenvs/airflow_cli_test/lib/python3.7/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)
[2019-08-13 22:30:04,876] {__init__.py:51} INFO - Using executor LocalExecutor
```

**Example implementation**:
![image](https://user-images.githubusercontent.com/8811558/62978759-eeb7e600-be19-11e9-89d5-ada08536adae.png)


If we then run `eval $(airflow autocomplete)` it gives the following error:
```
/Users/kaxilnaik/Documents/GitHub/incubator-airflow/airflow/models/dagbag.py:21: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
/Users/kaxilnaik/.virtualenvs/airflow_cli_test/lib/python3.7/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)
zsh: bad pattern: [2019-08-13
```

as it is trying to parse the log line `[2019-08-13 22:29:39,045] {settings.py:173} INFO - settings.configure_orm(): Using pool settings. pool_size=5, max_overflow=10, pool_recycle=1800, pid=86465`

Hence, not implementing this as part of this PR. Let's revisit this later.

Reply:
Oh that damn logging whenever we `import airflow` :(